etna/transforms/encoders/categorical.py

```diff
@@ -40,8 +57,6 @@ def transform(self, y: pd.Series, strategy: str):
             filling_value = -1
         elif strategy == ImputerMode.mean:
             filling_value = np.mean(encoded[~np.isin(y, diff)])
-        else:
```

Comment:
It seems to me that `strategy doesn't have a `str` type anymore, we should change that.

Same comment about `assert_never`.